packages/react/src/utility-components/AnimateHeight/AnimateHeight.tsx

```diff
@@ -20,7 +20,12 @@ export const AnimateHeight = ({
   const [height, setHeight] = useState<number>(0);
   const contentRef = useCallback(
     (node: HTMLDivElement) => {
-      setHeight(node && open ? node.getBoundingClientRect().height : 0);
+      if (node) {
+        const resizeObserver = new ResizeObserver(() => {
+          setHeight(open ? node.getBoundingClientRect().height : 0);
+        });
+        resizeObserver.observe(node);
```

Comment:
Do we need to worry about cleaning up? Unobserving the ResizeObserver 🤔

Reply:
Should not be a problem though? The node is in this component, so when the component is no longer in use its removed from the dom anyway.

Reply:
Exactly, I don't think it's necessary.